client/src/components/Explore.js

```diff
@@ -1,102 +1,142 @@
+    margin: "2rem 0 1rem 0",
+    textAlign: "center",
+  },
+  buttonContainer: {
+    marginBottom: "2rem",
+  },
+  button: theme.buttonPrimary,
 }));
 
-function Explore() {
-	//To get image for exploreCard, make sure the image is saved in public with the name of the place. ex: "public/images/cancun.png"
-	//THIS: props.location.toLowerCase() === saved image name
-	const theme = useTheme();
-	const classes = exploreStyles(theme);
+function Explore(props) {
+  //To get image for exploreCard, make sure the image is saved in public with the name of the place. ex: "public/images/cancun.png"
+  //THIS: props.location.toLowerCase() === saved image name
+  const { allLocations } = props;
+  const theme = useTheme();
+  const classes = exploreStyles(theme);
+  const [state, setState] = useState({
+    locations: [],
+    favorites: [],
+  });
 
-	const locations = [
-		{ location: "Cancun", country: "Mexico", imgName: "/cancun.png" },
-		{ location: "London", country: "UK", imgName: "/london.png" },
-		{ location: "Bali", country: "Indonesia", imgName: "bali.png" },
-		{ location: "Oslo", country: "Norway", imgName: "oslo.png" },
-		{ location: "Paris", country: "France", imgName: "paris.png" },
-		{ location: "New York", country: "USA", imgName: "newyork.png" },
-		{ location: "Miami", country: "USA", imgName: "miami.png" },
-		{ location: "Rome", country: "Italy", imgName: "rome.png" },
-	];
-
-	return (
-		<>
-			<Grid
-				container
-				justify="center"
-				alignItems="center"
-				direction="row"
-				className={classes.pageContainer}
-			>
-				<Grid
-					container
-					item
-					xs={12}
-					justify="center"
-					alignItems="center"
-					className={classes.pt4}
-				>
-					<Grid item xs={12} container justify="center" alignItems="center">
-						<Typography variant="h2" className={classes.textAlign}>
-							Explore Destinations
-						</Typography>
-					</Grid>
-				</Grid>
-				<Grid item>
-					<Typography variant="h5" className={classes.headerSpacing}>
-						World's Top Destinations to Explore
-					</Typography>
-				</Grid>
-				<Grid
-					container
-					item
-					justify="center"
-					alignItems="center"
-					spacing={5}
-					className={classes.cardContainer}
-				>
-					{locations.map((loc) => {
-						return (
-							<Grid item xs={12} lg={3}>
-								<ExploreCard
-									location={loc.location}
-									country={loc.country}
-									imgName={loc.imgName}
-								></ExploreCard>
-							</Grid>
-						);
-					})}
-				</Grid>
-			</Grid>
-		</>
-	);
+  const shuffleLocations = async () => {
+    const favoriteData = await axios.get("/favorites/getAllFavorites");
+    const favoritesArray = favoriteData.data;
+    let filteredLocations = allLocations;
+    console.log([].findIndex((e) => e === 1));
+    if (favoritesArray.length > 0) {
+      filteredLocations = allLocations.filter(
+        (location) =>
+          favoritesArray.findIndex((fav) => fav.id === location.id) < 0
+      );
+    }
```

Comment:
This probably will not matter much in practice performance wise, but a clean way to do this is to make a set of the favourites
```
favorites = new Set();
favoritesArray.forEach(location => favorites.add(location.id);
filteredLocations = allLocations.filter(location => favorites.has(location.id));
```
This way works fine though so you don't have to change it